### PR TITLE
Introduce `CASSANDRA2` version store type, deprecate `CASSANDRA` version store type

### PR DESCRIPTION
**Pull request by snazy**
The current `CASSANDRA` version store type uses a lot of columns in the `objs` table, which cause quite some overhead in storage. The new `CASSANDRA2` version store type works like `CASSANDRA`, but uses a single column for all object values and also serializes object-ids as binary values.

### CODE REVIEW

**Comment by dimas-b** [comment]
Is this worth the trouble given that `CASSANDRA` is "experimental" still?

**Comment by snazy** [comment]
> Is this worth the trouble given that `CASSANDRA` is "experimental" still?

It's more about removing all the obj-type specific (de)serializers

File: servers/quarkus-config/src/main/java/org/projectnessie/quarkus/config/VersionStoreConfig.java [hunk 1]
```diff
@@ -30,7 +30,10 @@ enum VersionStoreType {
     ROCKSDB,
     DYNAMODB,
     MONGODB,
+    /** Cassandra variant using many distinct columns. */
     CASSANDRA,
```
Comment (dimas-b) [inline servers/quarkus-config/src/main/java/org/projectnessie/quarkus/config/VersionStoreConfig.java:34]: `@Deprecated` (as other old store types)?

File: versioned/storage/cassandra2/src/main/java/org/projectnessie/versioned/storage/cassandra2/Cassandra2BackendFactory.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.storage.cassandra2;
+
+import jakarta.annotation.Nonnull;
+import org.projectnessie.versioned.storage.common.persist.BackendFactory;
+
+public class Cassandra2BackendFactory implements BackendFactory<Cassandra2BackendConfig> {
+
+  public static final String NAME = "Cassandra";
```
Comment (dimas-b) [inline versioned/storage/cassandra2/src/main/java/org/projectnessie/versioned/storage/cassandra2/Cassandra2BackendFactory.java:23]: `Cassandra2`?

**Review by dimas-b — APPROVED** [review]

**Pull request merged**